### PR TITLE
Include band-related features (e.g. band center, bandwidth, skewness, kurtosis) in VASP schema

### PR DESCRIPTION
**Pull request by arosen93**
## Summary
Motivated by my [recent PR](https://github.com/materialsproject/pymatgen/pull/2455) in pymatgen, I thought it'd be nice to include some DOS-related properties in the VASP schema by default. Since the DOS filesizes are large, these descriptors are nice to have available. It includes the _s_-, _p_-, _d_-, and _f_-band filling, center, bandwidth, skewness, kurtosis, and upper edge for each unique element in the structure.

I'll let this sit as WIP for now until the pymatgen version gets updated and incorporated in Atomate2. Then I'll complete this, assuming that you are not against this feature.

For context, here is what it would look like:
```
{'Na': {'s': {'filling': 0.33120586862464046,
   'center': 1.7422969706881928,
   'bandwidth': 3.102142959545026,
   'skewness': -0.2167067390936569,
   'kurtosis': 2.1425007370548816,
   'upper_edge': 5.961509659999999}},
 'Zn': {'s': {'filling': 0.5890520285762735,
   'center': 0.08945415449633912,
   'bandwidth': 2.924298105584621,
   'skewness': 0.675656214195633,
   'kurtosis': 3.5493177947470667,
   'upper_edge': -0.8588903399999999},
  'p': {'filling': 0.10612463151169614,
   'center': 2.3106410963491792,
   'bandwidth': 1.9943166963316488,
   'skewness': -0.8375372113795722,
   'kurtosis': 12.889078546585981,
   'upper_edge': 3.64950966},
  'd': {'filling': 0.9216081624528645,
   'center': -1.1993643736637292,
   'bandwidth': 1.5116949803584443,
   'skewness': 3.530018103490887,
   'kurtosis': 16.476657421131446,
   'upper_edge': -1.43689034}}
}
```

### CODE REVIEW

**Comment by utf** [comment]
This looks great, thanks!

File: src/atomate2/vasp/schemas/calculation.py [hunk 1]
```diff
@@ -443,6 +449,48 @@ def from_vasp_outputs(
             magmoms = [m["tot"] for m in outcar.magnetization]
             structure.add_site_property("magmom", magmoms)
 
+        # Parse DOS properties
+        if hasattr(vasprun, "complete_dos"):
```
Comment (utf) [inline src/atomate2/vasp/schemas/calculation.py:453]: Please could you refactor this to a separate function, just to keep `from_vasp_outputs` from growing too large.

**Review by utf — COMMENTED** [review]
Hi @arosen93, I just had one minor comment.

File: src/atomate2/vasp/schemas/calculation.py (hunk 1 above)
Comment (arosen93) [reply src/atomate2/vasp/schemas/calculation.py:453]: Certainly. That makes sense! I will make that change when I write up the unit tests sometime in the next few days, now that the pymatgen version has been updated.

Comment (arosen93) [reply src/atomate2/vasp/schemas/calculation.py:453]: Resolved in e1b229d.

**Comment by arosen93** [comment]
@utf, would you be able to help me understand the `mypy` linting issue? I don't quite get why it wants a typehint for a variable that is not a arg/kwarg.

**Comment by utf** [comment]
Sorry about the linting issue it is related to this: https://github.com/psf/black/issues/2964

I think it should be fixed if you update the black version in pre-commit to 22.3.0

**Comment by utf** [comment]
If you pull the latest changes from master that should fix the linting error.

File: src/atomate2/vasp/schemas/calculation.py [hunk 2]
```diff
@@ -443,6 +449,10 @@ def from_vasp_outputs(
...
+            dosprop_dict = _get_band_props(vasprun.complete_dos, structure)
```
Comment (utf) [inline src/atomate2/vasp/schemas/calculation.py:454]: Do you need to set a default value for dosprop_dict in case there isn't a complete dos?

**Comment by arosen93** [comment]
@utf, thanks! That fixes the `black` issue. That said, I was actually wondering about the unrelated `mypy` issue. The error is `src/atomate2/vasp/schemas/calculation.py:791: error: Need type annotation for "dosprop_dict" (hint: "dosprop_dict: Dict[<type>, <type>] = ...")`. Of course, I can just go ahead and add the typehint, but I don't really understand _why_ it wants a typehint there since it's not a function kwarg/arg.

Planning to add unit tests soon, and then this is good to go.

**Comment by arosen93** [comment]
Okay, should be ready to go.

**Comment by utf** [comment]
Great thanks!

**Pull request merged**